### ISSUE TITLE
Add MLFlowLogger

### ISSUE THREAD

**Issue by deepslug** [OML-Team/open-metric-learning#509]
I've just started using OML and I absolutely love it! (More questions are on the way...)

It'd be nice if it supported MLFlowLogger.
https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.MLFlowLogger.html

**Comment by Fissium** [comment]
@deepslug Use can use the following code example to add the mlflow logger to your pipeline:

```python
import logging
import warnings

import hydra
from omegaconf import DictConfig
from oml.const import HYDRA_VERSION
from oml.lightning.pipelines.train import extractor_training_pipeline
from oml.registry.loggers import LOGGERS_REGISTRY
from pytorch_lightning.loggers import MLFlowLogger

logging.basicConfig(level=logging.INFO)

logger = logging.getLogger(__name__)

warnings.filterwarnings("ignore")


LOGGERS_REGISTRY["mlflow"] = MLFlowLogger


@hydra.main(
    config_path="./configs/",
    config_name="train_vit_mlflow.yaml",
    version_base=HYDRA_VERSION,
)
def main_hydra(cfg: DictConfig) -> None:
    logger.info("Training model.")
    extractor_training_pipeline(cfg)


if __name__ == "__main__":
    main_hydra()
```

Then add something similar to your config:

```yaml
logger:
  name: mlflow
  args:
    experiment_name: "ViT_oml"
    run_name: "vit_log_artifacts"
```

**Comment by AlekseySh** [comment]
@deepslug Thank you for using OML :) Waiting for your further questions

@Fissium Thank you so much for such a great answer! It's a valid approach, but it requires changing source code of the library (as you mentioned in the "final step"). To avoid this and make it available from pip, we need someone who can add this functionality to the repo. So, contributions are welcome :)

**Comment by deepslug** [comment]
@Fissium 
Thank you for your detailed answer! I've actually tried something similar and realized I had to modify parser.py, hence my question.

@AlekseySh 
I'll post my question when I have everything sorted out.

**Comment by AlekseySh** [comment]
@deepslug , great, thank you

**Comment by AlekseySh** [comment]
@armored-guitar is going to work on this issue.
